stdlib/internal/promql/join.go

```diff
@@ -0,0 +1,553 @@
+package promql
+
+import (
+	"context"
+	"sort"
+	"sync"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/compiler"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/interpreter"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+const joinKind = "promJoin"
```

Comment:
Can you name this with the full pathname? We're going to run into this issue more often and we should set a standard for this. Something like `internal/promql.join`?